linera-views/src/value_splitting.rs

```diff
@@ -0,0 +1,378 @@
+            let big_key = big_key?;
+            let len = big_key.len();
+            if Self::read_index_from_key(big_key)? == 0 {
+                let key = big_key[0..len - 4].to_vec();
+                keys.push(key);
+            }
+        }
+        Ok(keys)
+    }
+
+    async fn find_key_values_by_prefix(
+        &self,
+        key_prefix: &[u8],
+    ) -> Result<Self::KeyValues, Self::Error> {
+        let mut key_values = Vec::new();
+        let mut key = Vec::new();
+        let mut big_value = Vec::new();
+        let mut count = None;
+        let mut pos = 0;
+        for result in self
+            .client
+            .find_key_values_by_prefix(key_prefix)
+            .await?
+            .iterator()
+        {
+            let (big_key, value) = result?;
+            let len = big_key.len();
+            let idx = Self::read_index_from_key(big_key)?;
+            if idx == 0 {
+                count = Some(Self::read_count_from_value(value)?);
+                key = big_key[..len - 4].to_vec();
+                big_value.extend(value[4..].to_vec());
+                pos = 1;
+            } else if big_key[..len - 4] == key {
+                if let Some(count) = count {
+                    if idx == pos && idx < count {
+                        big_value.extend(value);
+                        pos += 1;
+                    }
+                }
+            }
+            if let Some(count_pos) = count {
+                if pos == count_pos {
+                    key_values.push((mem::take(&mut key), mem::take(&mut big_value)));
+                    count = None;
+                }
+            }
+        }
+        Ok(key_values)
+    }
+
+    async fn write_batch(&self, batch: Batch, base_key: &[u8]) -> Result<(), Self::Error> {
+        let mut batch_new = Batch::new();
+        for operation in batch.operations {
+            match operation {
+                WriteOperation::Delete { key } => {
+                    let mut big_key = key.to_vec();
+                    big_key.extend(&[0, 0, 0, 0]);
+                    batch_new.delete_key(big_key);
+                }
+                WriteOperation::Put { key, value } => {
+                    let big_key = Self::get_segment_key(&key, 0)?;
+                    let mut first_chunk = Vec::new();
+                    let mut pos: u32 = 0;
+                    for value_chunk in value.chunks(K::MAX_VALUE_SIZE) {
+                        let big_key_segment = Self::get_segment_key(&key, pos)?;
+                        if pos == 0 {
+                            first_chunk = value_chunk.to_vec();
+                        } else {
+                            batch_new.put_key_value_bytes(big_key_segment, value_chunk.to_vec());
+                        }
+                        pos += 1;
+                    }
+                    let count = if pos == 0 { 1 } else { pos };
+                    let value_ext = Self::get_initial_count_first_chunk(count, &first_chunk)?;
```

Comment:
Won't this be too big? `K::MAX_VALUE_SIZE + 4`?

Reply:
Yes, solved.